### PR TITLE
pow2: Adding monotonicity lemma 

### PR DESCRIPTION
**Pull request by yoni206**
We add the lemma x<=y --> pow2(x)<=pow2(y) to the pow2 solver.
Additionally, some renaming of variables is introduced for better clarity.

### CODE REVIEW

**Review by ajreynol — CHANGES_REQUESTED** [review]
You are using a quadratic algorithm that sends redundant lemmas :)  Do write this as using a sort algorithm; then the lemmas are linear.  i.e. pow2(t1) .... pow2(tn), sort by t1^M .... tn^M then only add n-1 lemmas C => pow2(t_i1) < pow2( t_i2 ) .... C => pow2( t_i{n-1}) < pow2( t_in ).

File: src/theory/arith/nl/pow2_solver.cpp [hunk 1]
```diff
@@ -90,31 +90,68 @@ void Pow2Solver::checkFullRefine()
 {
   Trace("pow2-check") << "Pow2Solver::checkFullRefine";
   Trace("pow2-check") << "pow2 terms: " << std::endl;
-  for (const Node& i : d_pow2s)
+  NodeManager* nm = NodeManager::currentNM();
```
Comment (ajreynol) [inline src/theory/arith/nl/pow2_solver.cpp:93]: Can you move this to its own method?

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/theory/arith/nl/pow2_solver.cpp (hunk 1 above)
Comment (yoni206) [reply src/theory/arith/nl/pow2_solver.cpp]: done

**Review by ajreynol — CHANGES_REQUESTED** [review]
Would be good to have a regression to test the code works already, or is this not possible yet?

**Comment by yoni206** [comment]
Regressions are 1 PR away, if that's ok with you

**Review by ajreynol — APPROVED** [review]

**Pull request merged**